WorkflowReactiveSwift/Sources/SignalProducerWorkflow.swift

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import ReactiveSwift
+import Workflow
+import class Workflow.Lifetime
+
+extension SignalProducer: AnyWorkflowConvertible where Error == Never {
+    public func asAnyWorkflow() -> AnyWorkflow<Void, Value> {
+        return SignalProducerWorkflow(signalProducer: self).asAnyWorkflow()
+    }
+}
+
+struct SignalProducerWorkflow<Value>: Workflow {
+    public typealias Output = Value
+    public typealias State = Void
+    public typealias Rendering = Void
+
+    var signalProducer: SignalProducer<Value, Never>
+    var key: AnyHashable
+
+    public init(signalProducer: SignalProducer<Value, Never>, key: AnyHashable = "") {
+        self.signalProducer = signalProducer
+        self.key = key
```

Comment:
I don't think this `key` makes any sense, right? There'll always be just one `sideEffect` in this workflow. cc: @bencochran

Reply:
Yeah, agreed. Consumers can provide a key when running this workflow itself.